src/components/IOUConfirmationList.js

```diff
@@ -102,21 +112,66 @@ const defaultProps = {
     comment: '',
     network: {},
     myPersonalDetails: {},
+    iouType: CONST.IOU.IOU_TYPE.REQUEST,
 };
 
 class IOUConfirmationList extends Component {
     constructor(props) {
         super(props);
 
-        this.toggleOption = this.toggleOption.bind(this);
-
         const formattedParticipants = _.map(this.getParticipantsWithAmount(this.props.participants), participant => ({
             ...participant, selected: true,
         }));
 
+        // Add the button options to payment menu
+        const confirmationButtonOptions = [];
+        let defaultButtonOption = {
+            text: this.props.translate(this.props.hasMultipleParticipants ? 'iou.split' : 'iou.request', {
+                amount: this.props.numberFormat(
+                    this.props.iouAmount,
+                    {style: 'currency', currency: this.props.iou.selectedCurrencyCode},
+                ),
+            }),
+        };
+        if (this.props.iouType === CONST.IOU.IOU_TYPE.SEND && this.props.participants.length === 1) {
+            // Add the Expensify Wallet option if available and make it the first option
+            if (this.props.localCurrencyCode === CONST.CURRENCY.USD || Permissions.canUsePayWithExpensify(this.props.betas)) {
+                confirmationButtonOptions.push({text: this.props.translate('iou.settleExpensify'), icon: Wallet});
+            }
+
+            // Add PayPal option
+            if (this.props.participants[0].payPalMeAddress) {
+                confirmationButtonOptions.push({text: this.props.translate('iou.settlePaypalMe'), icon: PayPal});
+            }
+            defaultButtonOption = {text: this.props.translate('iou.settleElsewhere'), icon: Cash};
+        }
+        confirmationButtonOptions.push(defaultButtonOption);
+
+        this.isComponentMounted = false;
+
         this.state = {
+            confirmationButtonOptions,
             participants: formattedParticipants,
         };
+
+        this.toggleOption = this.toggleOption.bind(this);
+        this.onPress = this.onPress.bind(this);
+    }
+
+    componentDidMount() {
+        this.isComponentMounted = true;
```

Comment:
Apparently this is an antipattern (that I haven't run into before) that should be avoided https://reactjs.org/blog/2015/12/16/ismounted-antipattern.html

Based on the docs I think the solution is to make `addVenmoPaymentOption` a promise and cancel it in `componentWillUnmount`

Reply:
> Based on the docs I think the solution is to make addVenmoPaymentOption a promise and cancel it in componentWillUnmount

cool! I'll need to take a look into this

Reply:
Ah, seems like the antipattern is using the method `isMounted()` and they actually recommend the solution I went with as a simple way to get the job done:

> An easy migration strategy for anyone upgrading their code to avoid isMounted() is to track the mounted status yourself. Just set a _isMounted property to true in componentDidMount and set it to false in componentWillUnmount, and use this variable to check your component’s status.

Reply:
It sound to me like that is an easy migration strategy for people who are already using `isMounted()` and don't want their code to straight up break. The next paragraph talks about how you should actually fix it because it is still an antipattern where we aren't cleaning up properly, you just aren't using the method anymore.

Reply:
I've implemented the cancelable promise, but taking a look at the original [deprecation thread](https://github.com/facebook/react/issues/5465) there seems to be a good bit of back and forth on whether it's the best approach.

We do use the `isComponentMounted` flag in a few other places in our code, so I'd like to clarify in #eng-chat if it's worth to move forward with it or to keep as is.

Reply:
@stitesExpensify from the [Slack thread](https://expensify.slack.com/archives/C03TQ48KC/p1632428818222500) it looks like we have two options: a) do it or b) as Marc said:

>If we are already using the isComponentMounted  strategy and it works I’d stick with that for consistency. But maybe create a follow up issue to discuss / change all the usages. Not really seeing why one would be much better than the other tho (probably missing something).

I'm leaning towards b) but am curious to know what you think.

Reply:
Looks like we settled on a) so will re-test and push those changes